spec/features/plp_spec.rb

```diff
@@ -0,0 +1,196 @@
+          max_price = price.split(' ')[1].delete('$')
+          sorted_hash = products_prices.select { |key, value| value <= max_price.to_f }
+        elsif price.include?('over')
+          min_price = price.split(' ')[0].delete('$')
+          sorted_hash = products_prices.select { |key, value| value >= min_price.to_f }
+        else
+          range = price.gsub('$', '').delete('-').split(' ')
+          sorted_hash = products_prices.select { |key, value| value >= range[0].to_f && value <= range[1].to_f }
+        end
+
+        check price
+        find('.btn-primary').click
+
+        aggregate_failures do
+          if !sorted_hash.empty?
+            sorted_products = all('span.info').collect(&:text)
+
+            expect(sorted_hash.keys).to eq sorted_products
+          else
+            expect(page).to have_css('[data-hook="products_search_results_heading"]', text: 'No products found')
+          end
+        end
+
+        visit router.rails_category_path
+      end
+    end
+  end
+
+  context 'starting with Admin Panel' do
+    before do
+      admin_login
+      visit router.admin_products_path
+    end
+
+    it 'has all products' do
+      products = all('tr[data-hook="admin_products_index_rows"]', text: 'Available')
+
+      pictures = products.collect { |product| product.find('img')[:src].split('/').last }
+      titles   = products.collect { |product| product.all('td')[3].text }
+      prices   = products.collect { |product| product.all('td')[4].text }
+
+      visit '/'
+
+      titles.each_with_index do |title, index|
+        until page.has_css?('.panel-default', text: title)
+          raise "The product #{title} is not found" unless page.has_css?('a', text: 'Next')
+          click_link 'Next'
+        end
+
+        pdp_product = find('.panel-default', text: title)
+        pdp_picture = pdp_product.find('img')[:src].split('/').last
+
+        aggregate_failures do
+          expect(pdp_product.text).to include prices[index]
+          expect(pdp_picture).to eq pictures[index]
+        end
+
+        visit '/'
+      end
+    end
+
+    it 'has all available products from Admin Panel' do
+      quantity = all('tr[data-hook="admin_products_index_rows"]', text: 'Available').count
+
+      visit '/'
+
+      all('.page').count.times do
+        quantity_per_page = all('.product-body').count
+        quantity -= quantity_per_page
+
+        click_link 'Next' if page.has_css?('.next_page')
+      end
+
+      expect(quantity).to eq 0
+    end
+
+    it 'can search for a product' do
+      products = all('tr[data-hook="admin_products_index_rows"]', text: 'Available')
+      titles = products.collect { |product| product.all('td')[3].text }
+
+      visit '/'
+
+      titles.each do |title|
+        fill_in 'Search', with: title
+        click_button 'Search'
+
+        products_titles = all('.product-body').collect(&:text)
+
+        products_titles.each do |all_words|
+          tag = all_words.split(' ')
```

Comment:
No need to specify `' '` as a separator since it's a default behavior of split, e.g `hey hey.split #=> ['hey', 'hey']`